File: war/src/test/js/pages/login_page.js
'use strict';

var LoginPage = function() {
    this.userName = element(by.binding('ctrl.user.displayName'));

    this.isLoggedIn = function() {
	return this.userName.isDisplayed();
    }

    this.currentUser = function() {
	return this.userName.getText();
    }

    this.login = function(username, password) {
	return this.isLoggedIn().then(function(loggedIn) {
	    if (loggedIn) {
		element(by.id('logout_link')).click().then(function() {
		    console.log("User is logged in, Forcing a logout");
		});
	    }
	}).then(function() {
	    browser.waitForAngular();
	    console.log("log in as ", username);
	    element(by.id('login_link')).click();
	    element(by.id('login_username')).sendKeys(username);
	    element(by.id('login_password')).sendKeys(password);
	    return element(by.id('loginSubmit')).click();
	})

    };

    this.logout = function() {
	return this.isLoggedIn().then(function(loggedIn) {
	    if (loggedIn) {
		console.log('logging out')
		return element(by.id('logout_link')).click();
	    }
	});
    };

};

module.exports = LoginPage;

File: war/src/test/java/no/nav/aura/basta/it/pages/LoginPage.java
package no.nav.aura.basta.it.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class LoginPage {

	private WebDriver driver;

	public LoginPage(WebDriver driver, String url) {
		this.driver = driver;
		driver.get(url);
	}


	public void login(String user, String password) {
        WebElement loginLink = driver.findElement(By.id("login_link"));
        WebElement logoutLink = driver.findElement(By.id("logout_link"));
        loginLink.click();
		WebElement loginForm = driver.findElement(By.id("loginForm"));
		WebElement usernameInput = loginForm.findElement(By.id("login_username"));
		WebElement passwordInput = loginForm.findElement(By.id("login_password"));
		usernameInput.sendKeys(user);
		passwordInput.sendKeys(password);
		passwordInput.submit();
	}

	public boolean isLoggedIn() {
        WebElement currentUser = driver.findElement(By.id("currentUser"));
        System.out.println(currentUser.getText());
        return !currentUser.getText().isEmpty();
	}

}
